clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetchMetricsManager.java

```diff
@@ -101,32 +105,38 @@ void recordRecordsFetched(int records) {
 
     void recordBytesFetched(String topic, int bytes) {
         String name = topicBytesFetchedMetricName(topic);
-        Sensor bytesFetched = new SensorBuilder(metrics, name, () -> topicTags(topic))
-                .withAvg(metricsRegistry.topicFetchSizeAvg)
-                .withMax(metricsRegistry.topicFetchSizeMax)
-                .withMeter(metricsRegistry.topicBytesConsumedRate, metricsRegistry.topicBytesConsumedTotal)
-                .build();
+        maybeRecordDeprecatedBytesFetched(name, topic, bytes);
+
+        Sensor bytesFetched = new SensorBuilder(metrics, name, () -> Map.of("topic", topic))
+            .withAvg(metricsRegistry.topicFetchSizeAvg)
+            .withMax(metricsRegistry.topicFetchSizeMax)
+            .withMeter(metricsRegistry.topicBytesConsumedRate, metricsRegistry.topicBytesConsumedTotal)
+            .build();
         bytesFetched.record(bytes);
     }
 
     void recordRecordsFetched(String topic, int records) {
         String name = topicRecordsFetchedMetricName(topic);
-        Sensor recordsFetched = new SensorBuilder(metrics, name, () -> topicTags(topic))
-                .withAvg(metricsRegistry.topicRecordsPerRequestAvg)
-                .withMeter(metricsRegistry.topicRecordsConsumedRate, metricsRegistry.topicRecordsConsumedTotal)
-                .build();
+        maybeRecordDeprecatedRecordsFetched(name, topic, records);
+
+        Sensor recordsFetched = new SensorBuilder(metrics, name, () -> Map.of("topic", topic))
+            .withAvg(metricsRegistry.topicRecordsPerRequestAvg)
+            .withMeter(metricsRegistry.topicRecordsConsumedRate, metricsRegistry.topicRecordsConsumedTotal)
+            .build();
         recordsFetched.record(records);
     }
 
     void recordPartitionLag(TopicPartition tp, long lag) {
         this.recordsLag.record(lag);
 
         String name = partitionRecordsLagMetricName(tp);
-        Sensor recordsLag = new SensorBuilder(metrics, name, () -> topicPartitionTags(tp))
-                .withValue(metricsRegistry.partitionRecordsLag)
-                .withMax(metricsRegistry.partitionRecordsLagMax)
-                .withAvg(metricsRegistry.partitionRecordsLagAvg)
-                .build();
+        maybeRecordDeprecatedPartitionLag(name, tp, lag);
+
+        Sensor recordsLag = new SensorBuilder(metrics, name, () -> mkMap(mkEntry("topic", tp.topic()), mkEntry("partition", String.valueOf(tp.partition()))))
```

Comment:
Could we do the same for the deprecated metrics too for better consistency?

Reply:
Done.

Reply:
Is there a reason why you're still using the `Utils.mkMap` methods instead of `Map.of` from Java 9+?